tiny_dnn/layers/convolutional_layer.h

```diff
@@ -231,9 +232,17 @@ class convolutional_layer : public feedforward_layer<Activation> {
      * @param out_data     output vectors
      **/
     void forward_propagation(const std::vector<tensor_t*>& in_data,
-                             std::vector<tensor_t*>&       out_data) override {
+                             std::vector<tensor_t*>&       out_data) override { 
+        copy_and_pad_input(*in_data[0], cws_.prev_out_padded_);
+
+        std::vector<tensor_t*> in_data_;
+        in_data_.push_back(&cws_.prev_out_padded_);
+        for (cnn_size_t i = 1; i < in_data.size(); ++i) {
+            in_data_.push_back(in_data[i]);
+        }
+
```

Comment:
@nyanp
convolution tests are fixed with https://github.com/tiny-dnn/tiny-dnn/pull/298/commits/ca862cef3eb49a3487f743ea79cb0e864636d603. 
However, network tests which have convolution operations are broken. So, I suspect that the codes above are not a proper solution to share data between neighbor layers.